core/src/main/java/org/infinispan/LockedStream.java

```diff
@@ -84,6 +88,55 @@ default void forEach(SerializableBiConsumer<Cache<K, V>, ? super CacheEntry<K, V
    }
 
    /**
+    * Performs a function for each element of this stream on the primary owner of the given key returning
+    * a value. The returned value from the function will be sent back to the user mapped to the key that generated
+    * it, with all of these stored in a map. This method will return only after all entries have been processed.
+    * <p>
+    * This method is currently marked as {@link Experimental} since this method returns a Map and requires blocking.
```

Comment:
Whether the commands themselves block the caller is subjective to where they are executed. Currently this requires blocking because of the iterator. My plan is to eventually change this method to return a `Publisher` with appropriate back pressure through subscription requests available.